### ISSUE TITLE
SimDem2 Feature: Add Validation Mode

### ISSUE THREAD

**Issue by lastcoolnameleft** [Azure/simdem#90]
It would be useful to create a validation mode that would not run any commands, but instead check for the following:

* Missing environment variables
* Broken Links (in prereqs, next steps, etc.)

**Comment by SorraTheOrc** [comment]
+1

We should also add linting of the content to this (see #91)

**Comment by SorraTheOrc** [comment]
Observation: the word "validation" is now used in three different contexts, we need to have one meaning for each word.

The three uses of "Validation" at the time of writing are:
  1. Validation section: used to validate whether a prerequisite needs to run
  1. Validation of test results
  1. this proposed validation mode

**Comment by lastcoolnameleft** [comment]
You're right, that is getting conflated.  Personally, this is how I'd break it up: (however, it conflicts with SimDem1 compatibility)

* Validation Mode - A SimDem mode which checks if the markdown passes the linter, missing env vars, does not have broken links, etc.
* Verification - A section in Markdown which checks if a prerequisite needs to run
* Assertion - A section in Markdown which checks if the tests passes.

**Comment by SorraTheOrc** [comment]
That language works for me.

**Comment by lastcoolnameleft** [comment]
Cool.  Are you ok if we implement this into SimDem2?  It would break existing functionality in SimDem 1:

* `# Validation` section would need to be renamed to `# Verification`
* `Results:` would need to be renamed to `Assert:`

We could leave the code in there to check for both, but to reduce complexity and tech debt, I would like to make the transition as quickly as possible.

**Comment by SorraTheOrc** [comment]
I wasn't thinking of changing the .md structure for this, just the language we use. But I think there is a compromise to be had.

`# Validation` -> `# Verification`

No problem with this change. However, I'd prefer for "Validation" to continue to work (with a deprecation warning if you like). I could be convinced to soften on this, but it will make testing a little slower as I can only switch to SimDem2 once my live documents work within it. I'm happy to maintain a SImDem2 branch of my current project though if you prefer to make a clean cut now.

`Results:` should not change to Assert though. It seems "wrong" from a documentation author perspective. It does not read naturally. Since the word "Results" doesn't clash with "Validation" it's not part of the confusion I was raising here. I'd be OK with you making Assert and option (I see how in some kinds of documents it is actually better), but I'm not supportive of it being the only option.

Your call.
